polkadot/node/core/pvf/src/artifacts.rs

```diff
@@ -212,53 +365,49 @@ impl Artifacts {
 	pub fn insert_preparing(
 		&mut self,
 		artifact_id: ArtifactId,
-		waiting_for_response: Vec<PrepareResultSender>,
+		waiting_for_response: Vec<PrecheckResultSender>,
 	) {
 		// See the precondition.
 		always!(self
-			.artifacts
+			.inner
 			.insert(artifact_id, ArtifactState::Preparing { waiting_for_response, num_failures: 0 })
 			.is_none());
 	}
 
 	/// Insert an artifact with the given ID as "prepared".
-	///
-	/// This function must be used only for brand-new artifacts and should never be used for
-	/// replacing existing ones.
-	#[cfg(test)]
-	pub fn insert_prepared(
+	pub(crate) fn insert_prepared(
 		&mut self,
 		artifact_id: ArtifactId,
+		path: PathBuf,
 		last_time_needed: SystemTime,
 		prepare_stats: PrepareStats,
 	) {
 		// See the precondition.
```

Comment:
The precondition was dropped. Either this comment should be removed as well, or the precondition should be reintroduced (which makes sense, as it's implied by `always!` and it would be good to explain to a code reader why it always holds).